packages/bundler-plugin-core/src/utils/Output.ts

```diff
@@ -25,28 +42,76 @@ class Output {
   assets?: Asset[];
   chunks?: Chunk[];
   modules?: Module[];
-  plugin?: {
+  #internalPlugin?: {
     name: string;
     version: string;
   };
 
   constructor(userOptions: NormalizedOptions) {
     this.version = "1";
     this.userOptions = userOptions;
+    this.#internalBundleName = userOptions.bundleName;
   }
 
-  start(pluginName: string, pluginVersion: string) {
+  start() {
     this.builtAt = Date.now();
-    this.plugin = {
-      name: pluginName,
-      version: pluginVersion,
-    };
   }
 
   end() {
     this.duration = Date.now() - (this.builtAt ?? 0);
   }
 
+  setBundleName(bundleName: string, options: SetPluginOptions = {}) {
+    // if set to false, unfreeze the plugin details
+    if (typeof options.frozen === "boolean" && !options.frozen) {
+      this.#internalOptions.frozenBundleName = options.frozen;
+    }
+
+    if (!this.#internalOptions.frozenBundleName) {
+      this.#internalBundleName = bundleName;
+    }
+
+    // lock back up if frozen is set to true
+    if (typeof options.frozen === "boolean" && options.frozen) {
+      this.#internalOptions.frozenBundleName = options.frozen;
+    }
+
+    return this.#internalBundleName ?? this.userOptions.bundleName;
```

Comment:
bundleName should be valid here - so we don't need this anymore. Ditto with the `get bundleName`.

```suggestion
    return this.#internalBundleName;
```

Reply:
Ahh shoot sorry missed that 🤦